demos/browser/app/meetingV2/meetingV2.ts

```diff
@@ -1757,11 +1796,64 @@ export class DemoMeetingApp implements
       return DefaultDeviceController.synthesizeVideoDevice('smpte');
     }
 
-    if (value === 'None') {
-      return null;
+    return value;
+  }
+
+  private videoFilterToProcessor(videoFilter: string): VideoFrameProcessor | null {
+    this.log(`Choosing video filter ${this.videoFilter}`);
+
+    if (videoFilter === 'Emojify') {
+      return new EmojifyVideoFrameProcessor('🚀');
     }
 
-    return value;
+    if (videoFilter === 'CircularCut') {
+      return new CircularCut();
+    }
+
+    if (videoFilter === 'NoOp') {
+      return new NoOpVideoFrameProcessor();
+    }
+
+    return null;
+  }
+
+  private chosenDevice: DefaultVideoTransformDevice;
+  private chosenProcessor: string;
+
+  private async stopChosenTransformDevice(): Promise<void> {
+    await this.chosenDevice?.stop();
+    this.chosenDevice = null;
+  }
+
+  private async videoInputSelectionWithOptionalFilter(innerDevice: Device): Promise<VideoInputDevice> {
+    if (this.videoFilter === 'None') {
+      return innerDevice;
+    }
+    
+    if (this.chosenDevice && this.videoFilter === this.chosenProcessor) {
```

Comment:
You have a bit of a collision here between `videoFilter` (the one selected in the dropdown) and `chosen*` (the one that's active).

I suggest renaming `videoFilter` to `selectedVideoFilterItem`, and using `chosenVideoFilter` for the latter.

Reply:
👍